cdap-ui/app/cdap/components/Replicator/Create/Content/Assessment/TablesAssessment/WithIssuesTables/index.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
+import { ITable } from 'components/Replicator/Create/Content/Assessment/TablesAssessment';
+import If from 'components/If';
+
+const styles = (theme): StyleRules => {
+  return {
+    root: {
+      height: '100%',
+    },
+    text: {
+      color: theme.palette.grey[100],
+      marginBottom: '10px',
+    },
+    noTablesText: {
+      fontWeight: 600,
+      padding: '5px 7px',
+      borderBottom: `1px solid ${theme.palette.grey[500]}`,
+    },
+    gridWrapper: {
+      height: '100%',
+      '& .grid.grid-container.grid-compact': {
+        maxHeight: '300px',
```

Comment:
I am seeing the grid pattern repeated. Can we extract out this style? Is this is something common across replicator?
It is going to be a nightmare when we want to replace this with the table component we create

Reply:
the difference is only the column template.. the problem is, it's hard to overwrite that since this css in js is have nested structure

Reply:
Can it be a generic function that takes in an input and based on that returns the specific styles?